### PR TITLE
sriov pkg: added sriovNetwork unit tests

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/sriov/network_test.go [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+
+		var testSettings *clients.Settings
+
+		testNetwork := generateNetwork(testCase.networkName, testCase.networkNamespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testNetwork)
+		}
+
+		testSettings = nil
```
Comment (sebrandon1) [inline pkg/sriov/network_test.go:96]: I believe this isn't needed since its just declared as a pointer above.

File: pkg/sriov/network_test.go [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+			assert.Equal(t, testNetwork.Name, builderResult.Object.Name)
+			assert.Equal(t, testNetwork.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestNewNetworkBuilder(t *testing.T) {
+	generateNetworkBuilder := NewNetworkBuilder
+
+	var testSettings *clients.Settings
```
Comment (sebrandon1) [inline pkg/sriov/network_test.go:124]: Move this inside the loop.

File: pkg/sriov/network_test.go [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+// buildValidTestBuilder returns a valid Builder for testing purposes.
+func buildInvalidSrIovNetworkTestBuilder(apiClient *clients.Settings) *NetworkBuilder {
+	return NewNetworkBuilder(
+		apiClient, defaultNetName, defaultNetNsName, defaultNetTargetNsName, "")
+}
+
+func buildTestClientWithDummyObject() *clients.Settings {
+	runtimeObjects := buildDummySrIovNetworkObject()
+
+	return clients.GetTestClients(runtimeObjects)
```
Comment (sebrandon1) [inline pkg/sriov/network_test.go:540]:
```suggestion
	return clients.GetTestClients(buildDummySrIovNetworkObject())
```

**Review by sebrandon1 — COMMENTED** [review]
Some comments!

I think I like the idea of moving `testSettings := buildTestClientWithDummyObject()` inside the testCase loops so its scope is more readable.

**Review by achuzhoy — DISMISSED** [review]

File: pkg/sriov/network_test.go (hunk 1 above)
Comment (kononovn) [reply pkg/sriov/network_test.go:96]: +1

**Review by sebrandon1 — APPROVED** [review]

**Pull request merged**